logical/sanitize.go

```diff
@@ -60,3 +65,46 @@ type HTTPWrapInfo struct {
 	CreationTime    time.Time `json:"creation_time"`
 	WrappedAccessor string    `json:"wrapped_accessor,omitempty"`
 }
+
+type HTTPSysInjector struct {
+	Response *HTTPResponse
+}
+
+func (h HTTPSysInjector) MarshalJSON() ([]byte, error) {
+	j, err := json.Marshal(h.Response)
+	if err != nil {
+		return nil, err
+	}
+
+	// Fast path no data or empty data
+	if h.Response.Data == nil || len(h.Response.Data) == 0 {
+		return j, nil
+	}
+
+	// Marshaling a response will always be a JSON object, meaning it will
+	// always start with '{', so we hijack this to prepend necessary values
+
+	// Make a guess at the capacity, and write the object opener
+	buf := bytes.NewBuffer(make([]byte, 0, len(j)+len(h.Response.Data)*100))
```

Comment:
Assuming that the multiplier of 100 is a random guess, I have another idea. Please see if it fits the need.
Since `h.Response.Data` will already be encoded in `j`, the size of the duplicated data within the new buffer is bound by the size of `j`. 

So, the length of the []byte can be assumed to be bound by `len(j) *2`. Does that make sense to you?